### PR TITLE
Add Servo Component Schema and Definition

### PR DESCRIPTION
**Pull request by brentru**
This pull request adds a servo component definition and schema.

New fields (these should be exposed to the user on the modal for every servo component:
* `frequency`: The desired frequency of the PWM signal, defaults to 50Hz
* `minPulseWidth`: The minimum pulse width of the servo in uS, defaults to 500uS
* `maxPulseWidth`: The maximum pulse width of the servo in uS, defaults to 2500uS

Requires https://github.com/adafruit/Wippersnapper_Protobuf/pull/95 to be implemented first.

### CODE REVIEW

**Comment by brentru** [comment]
@lorennorman Could you clarify why the check is failing on https://github.com/adafruit/Wippersnapper_Components/runs/6782177203?check_suite_focus=true#step:7:28 ?

**Comment by lorennorman** [comment]
It's failing because you edited a file outside the component files (schema.json). It's not a real failure since you're a core team member and you meant to do that, this is to bring our attention to external contributors attempting funny-business.

I'm not sure if there's a better way to handle this, open to suggestions. But yeah don't hesitate to merge this despite the ❌

File: components/servo/schema.json
Comment (lorennorman) [inline components/servo/schema.json:27]: Are there reasonable minimums/maximums we could leverage for these?

**Review by lorennorman — APPROVED** [review]
It looks fine, very simple. Up to you whether you do the min/max change I mentioned.

**Comment by brentru** [comment]
Holding off merging and converting to a draft until we're ready to implement.

**Comment by brentru** [comment]
Note: The JSON definition within here and servo implementation in Arduino has changed a bit since this pull request was originally made. 

Will update this tomorrow and tag when ready for review

**Comment by brentru** [comment]
Marking as ready for review. Defined visualization as `slider-servo`. We'll want to use a slider component type. Instead of icons on the ends of the sliders, we'll have labels for the servo's `minPulseWidth` and `maxPulseWidth`

File: components/servo/schema.json
Comment (lorennorman) [inline components/servo/schema.json:28]: Looking back at this and I notice that `min/maxPulseWidth` are required, so these defaults will never be used. I'd remove either the requirement or the defaults as it's redundant and seems to be giving some information that it's really not. Or clarify what we're up to here? Do you want me to heed these defaults somewhere, under some conditions?

Comment (brentru) [reply components/servo/schema.json:28]: @lorennorman  Yes, we should require min/maxpulseWidth. But the defaults should be used as the default values. Am I missing something here?

Comment (lorennorman) [reply components/servo/schema.json:28]: > ...defaults should be used as the default values for the generic servo component

I'm assuming that's the `generic/definition.json` in this PR. See how it contains `minPulseWidth` and `maxPulseWidth`? Because those values are provided, they will be used instead of the defaults.

Furthermore, because those values are `required` they will ALWAYS be present. The defaults cannot EVER be used because the properties they contain defaults for are required. Does that make sense?

Comment (brentru) [reply components/servo/schema.json:28]: Got it - thanks! Removing the defaults in next push.

**Comment by lorennorman** [comment]
@brentru i changed the directory (and thus the component's id) from `generic` to `servo`, i want to keep those keys descriptive about what they actually are (all i see in the code/database some places is the word "generic")

**Pull request merged**
